### PR TITLE
added a more centralized way to preserve temp files for debugging

### PR DESCRIPTION
**Pull request by notestaff**
Added a way to preserve tempfiles for debugging by setting an environment variable.  (This was part of PR #656 is-spades-2, but orthogonal to that PR so factoring it out here.)

### CODE REVIEW

File: util/file.py [hunk 1]
```diff
@@ -111,9 +110,29 @@ def tempfnames(suffixes, *args, **kwargs):
...
+            shutil.rmtree(name)
+
+def keep_tmp():
+    """Whether to preserve temporary directories and files (useful during debugging).
+    Return True if the environment variable VIRAL_NGS_TMP_DIRKEEP is set.
+    """
+    return 'VIRAL_NGS_TMP_DIRKEEP' in os.environ
 
 def set_tmp_dir(name):
     proposed_prefix = ['tmp']
```
Comment (tomkinsc) [inline util/file.py]: Not directly related to this PR, but should we make use of [`util.cmd.find_tmp_dir()`](https://github.com/broadinstitute/viral-ngs/blob/a3079b281a2ae3ebfd56095b4c0eae6775ad4874/util/cmd.py#L216) here to propose the tmp prefix?

**Comment by yesimon** [comment]
What about `--tmp_dirKeep` ?

File: util/file.py (hunk 1 above)
Comment (notestaff) [inline util/file.py]: here=within set_tmp_dir?

**Comment by notestaff** [comment]
@yesimon --tmp_dirKeep is a command-line option, not available when calling functions directly as opposed to calling CLI commands, hence the alternate way through the environment. Also, when debugging a snakemake pipeline, to turn keeptmp on/off you have to change every command in every rule; with the environment, you just set/unset one variable.

File: util/file.py (hunk 1 above)
Comment (tomkinsc) [inline util/file.py]: Nevermind, I see `find_tmp_dir` is used in `set_tmp_dir`. We could refactor to move `find_tmp_dir()` to`util.file`

**Pull request merged**